fuse/data/ops/ops_visprobe.py

```diff
@@ -76,111 +126,68 @@ def _extract_collected(self, sample_dict: NDict):
+            else :
+                 res[f'{prekey}.value'] = sample_dict[key].copy()
+            res[f'{prekey}.type'] = self._type_detector.get_type(sample_dict, key)
+            
+            if self._coco_converter != None :
+                 res[f'{prekey}.value'] = self._coco_converter(res[f'{prekey}.value'],res[f'{prekey}.type'])
+            if res[f'{prekey}.type'] in [DataTypeImaging.CRLE, DataTypeImaging.UCRLE  ]:
+                res[f'{prekey}.value'] = convert_COCO_to_mask(res[f'{prekey}.value'],sample_dict['height'],sample_dict['width'],res[f'{prekey}.type'] )
+                res[f'{prekey}.type'] = DataTypeImaging.SEG 
+            res[f'{prekey}.name'] = name
         return res
 
-
-    def _save(self, vis_data: Union[List, dict]):
-        # use caching to save all relevant vis_data
-        print("saving vis_data", vis_data)
-
-    def _handle_flags(self, flow, sample_dict: NDict, op_id: Optional[str]):
+    def _handle_flags(self, sample_dict: NDict, op_id: Optional[str]):
         """
         See super class
         """
         # sample was filtered out by its id
-        if self._id_filter and self.get_idx(sample_dict) not in self._id_filter:
+        if self._sample_id and self.get_idx(sample_dict) not in self._sample_id:
             return None
-        if flow not in self._flags:
-            return None 
        
-        # grouped key dictionary with the following structure:
-        #vis_data = {"cc_group":
-        #                {
-        #                    "key1": {
-        #                      "value": ndarray, 
-        #                      "type": DataType.Image,
-        #                      "op_id": "test1"}
-        #                    "key2": {
-        #                      "value": ndarray, 
-        #                      "type": DataType.BBox,
-        #                      "op_id": "test1"}
-        #                 },
-        #            "mlo_goup":
-        #                {
-        #                    "key3": {
-        #                      "value": ndarray, 
-        #                      "type": DataType.Image,
-        #                      "op_id": "test1"}
-        #                    "key4": {
-        #                      "value": ndarray, 
-        #                      "type": DataType.BBox,
-        #                      "op_id": "test1"}
-        #                 },
-        #            }
-        vis_data = self._extract_data(sample_dict, self._keys, op_id)
-        both_fr =  (VisFlag.REVERSE | VisFlag.FORWARD) in self._flags
-        dir_forward = flow == VisFlag.FORWARD
-        dir_reverse = flow == VisFlag.REVERSE
-        any_show_collected = VisFlag.SHOW_ALL_COLLECTED|VisFlag.SHOW_COLLECTED
-        
-        if VisFlag.COLLECT in self._flags or (dir_forward and both_fr):
+        vis_data = self._extract_data(sample_dict, self._keys ,self._name)
+        name_prefix=""
+        if "name" in sample_dict.to_dict().keys():
```

Comment:
Can you please remind me what is this name and where we set it?

Reply:
it's something I added , you can see in the example.
I wanted to have an option to give specific identifier to each image

Reply:
Can pass the name to OpProbe constructor instead?

Reply:
it already have name there, but it's a step name.
I need also a sample unique name but i'l use id